community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/LegacyStorePropertyCursor.java

```diff
@@ -0,0 +1,133 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api.store;
+
+import java.nio.channels.WritableByteChannel;
+
+import org.neo4j.kernel.api.cursor.PropertyCursor;
+import org.neo4j.kernel.impl.store.PropertyStore;
+import org.neo4j.kernel.impl.util.InstanceCache;
+
+/**
+ * Cursor for all properties on a node or relationship.
+ *
+ * This implementation uses the old PropertyBlock loading, which eager loads
+ * all data on init.
+ */
+public class LegacyStorePropertyCursor implements PropertyCursor
+{
+    private PropertyStore propertyStore;
+    private InstanceCache<LegacyStorePropertyCursor> instanceCache;
+    private PropertyRecordCursor propertyRecordCursor;
+    private PropertyBlockCursor propertyBlockCursor;
+
+    public LegacyStorePropertyCursor( PropertyStore propertyStore,
+            InstanceCache<LegacyStorePropertyCursor> instanceCache )
+    {
+        this.propertyStore = propertyStore;
+        this.instanceCache = instanceCache;
+    }
+
+    public LegacyStorePropertyCursor init( long firstPropertyId )
+    {
+        propertyRecordCursor = propertyStore.getPropertyRecordCursor( propertyRecordCursor, firstPropertyId );
+        propertyBlockCursor = null;
+        return this;
+    }
+
+    @Override
+    public boolean next()
+    {
+        if ( propertyBlockCursor != null && propertyBlockCursor.next() )
+        {
+            return true;
+        }
+        else if ( propertyRecordCursor.next() )
+        {
+            propertyBlockCursor = propertyRecordCursor.getPropertyBlockCursor( propertyBlockCursor );
+
+            return propertyBlockCursor.next();
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean seek( int keyId )
+    {
+        while ( next() )
+        {
+            if ( propertyBlockCursor.getPropertyBlock().getKeyIndexId() == keyId )
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    @Override
+    public int propertyKeyId()
+    {
+        return propertyBlockCursor.getPropertyBlock().getKeyIndexId();
+    }
+
+    @Override
+    public Object value()
+    {
+        return propertyBlockCursor.getProperty().value();
+    }
+
+    @Override
+    public boolean booleanValue()
+    {
+        return (Boolean) value();
+    }
+
+    @Override
+    public long longValue()
+    {
+        return ((Number)value()).longValue();
+    }
+
+    @Override
+    public double doubleValue()
+    {
+        return ((Number)value()).doubleValue();
+    }
+
+    @Override
+    public String stringValue()
+    {
+        return value().toString();
+    }
+
+    @Override
+    public void propertyData( WritableByteChannel channel )
+    {
+        throw new UnsupportedOperationException(  );
```

Comment:
This class is only retained for now (not used anywhere), so that if we find any problem with StorePropertyCursor we can easily switch back to old behaviour to compare. Will be removed eventually.